### ISSUE TITLE
Linting & Code Folding Don't Work

### ISSUE THREAD

**Issue by mrclary** [spyder-ide/mac-application#4]
As mentioned in #2, linting and code folding don't seem to be working in the standalone Mac application (SMA). These work as expected in from bootstrap startup and build/install-from-source.

I have saved the debug logs for three scenarios:
* start Spyder from bootstrap [spy38-bs-good.zip](https://github.com/spyder-ide/mac-application/files/4416656/spy38-bs-good.zip) -> works
* Start spyder from pip installed version (4.1.1) [spy38-good.zip](https://github.com/spyder-ide/mac-application/files/4416734/spy38-good.zip) -> works
* Start from SMA [spy38-sa-bad.zip](https://github.com/spyder-ide/mac-application/files/4416657/spy38-sa-bad.zip) -> doesn't work

These are attached as separate zip files. Each zip file contains the spyder debug log, the lsp logs, and requirements file (pip freeze). There are some differences between the lsp server logs that I noticed:
* Apparently no pyls plugins are loaded in the SMA and no plugins are reported as failed; some are loaded successfully and some fail to load in the bootstrap; all load successfully for the pip installed one.
* The pyls reports populated lists for linting and code folding in bootstrap and pip installed, but empty lists for SMA. This may be the result of plugins not being loaded.

Upon further investigation, in `pyls.config.config.py:51`, for SMA the `pkg_resources.iter_entry_points(PYLS)` generator yields no items.
```python
for entry_point in pkg_resources.iter_entry_points(PYLS):
    try:
        entry_point.load()
    except ImportError as e:
        log.warning("Failed to load %s entry point '%s': %s", PYLS, entry_point.name, e)
        self._pm.set_blocked(entry_point.name)
```
However, for bootstrap and pip install, the generator returns
```
autopep8 = pyls.plugins.autopep8_format
flake8 = pyls.plugins.flake8_lint
folding = pyls.plugins.folding
jedi_completion = pyls.plugins.jedi_completion
jedi_definition = pyls.plugins.definition
jedi_highlight = pyls.plugins.highlight
jedi_hover = pyls.plugins.hover
jedi_references = pyls.plugins.references
jedi_signature_help = pyls.plugins.signature
jedi_symbols = pyls.plugins.symbols
mccabe = pyls.plugins.mccabe_lint
preload = pyls.plugins.preload_imports
pycodestyle = pyls.plugins.pycodestyle_lint
pydocstyle = pyls.plugins.pydocstyle_lint
pyflakes = pyls.plugins.pyflakes_lint
pylint = pyls.plugins.pylint_lint
rope_completion = pyls.plugins.rope_completion
rope_rename = pyls.plugins.rope_rename
yapf = pyls.plugins.yapf_format
```

So...I think `pkg_resources` is not finding `pyls` plugins for SMA. I'm hoping there is way to solve this, but a posted issue (ronaldoussoren/py2app#256) is not encouraging.

Does anyone have any insight on this?

**Comment by mrclary** [comment]
Additionally, the SMA continuously starts, stops and restarts the language server.
<details><summary>Repeated Output Log</summary>

```
2020-04-01 16:41:45,934 [INFO] [spyder.plugins.completion.languageserver.plugin] -> Starting LSP client for python...
2020-04-01 16:41:45,934 [INFO] [spyder.plugins.completion.languageserver.client] -> Starting server: /Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/MacOS/python -m pyls --host 127.0.0.1 --port 2088 --tcp --check-parent-process -vv
2020-04-01 16:41:45,955 [INFO] [spyder.plugins.completion.languageserver.client] -> Starting transport: /Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/MacOS/python -u /Users/rclary/Documents/Python.git/developer/spyder-mac-app/dist/Spyder.app/Contents/Resources/lib/python3.8/spyder/plugins/completion/languageserver/transport/main.py --server-host 127.0.0.1 --server-port 2088  --folder /Users/rclary/.spyder-py3-dev/lsp_root_path --transport-debug 3 --external-server --zmq-in-port 64244 --zmq-out-port 64245
2020-04-01 16:41:45,976 [DEBUG] [spyder.plugins.completion.languageserver.client] -> LSP python client started!
2020-04-01 16:41:45,977 [INFO] [spyder.plugins.completion.languageserver.plugin] -> Restart successful!
2020-04-01 16:41:46,930 [INFO] [spyder.plugins.completion.languageserver.plugin] -> Automatic restart for python...
2020-04-01 16:41:51,930 [INFO] [spyder.plugins.completion.languageserver.plugin] -> Automatic restart attemp 1 for python...
2020-04-01 16:41:51,930 [INFO] [spyder.plugins.completion.languageserver.plugin] -> Stopping LSP client for python...
2020-04-01 16:41:51,930 [INFO] [spyder.plugins.completion.languageserver.client] -> Stopping python client...
```
</details>

**Comment by mrclary** [comment]
Okay. I've found a work-around for this issue.
`pkg_resources` is indeed not finding `pyls` plugins for SMA, although I'm still not sure why.

The good news is that I discovered that `pkg_resources` does find the `pyls`'s plugins if I copy the `.dist-info` and `.egg-info` from my site-packages into the applications Resources folder; go-figure. It seems a rather simple solution and perhaps crude, but it works. I don't know if there are any side-effects, though...

For completeness, I should mention that I stumbled onto pyinstaller/pyinstaller#3050, which provided a solution via monkey-patching `pkg_resources`. This worked for my MWE tests, however it will not work for our situations since `pyls` is invoked via a python subprocess and the monkey-patching won't carry through to the subprocess.
